lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb

```diff
@@ -30,13 +38,18 @@ def initialize(options = {})
 
     def realize_blob(file, _params)
       GoodData.gd_logger.info("Realizing download from Blob Storage. Container #{@container}.")
-      connect
-      filename = "#{SecureRandom.urlsafe_base64(6)}_#{Time.now.to_i}.csv"
-      blob_name = @path ? "#{file}" : "#{@path.delete_suffix('/')}/#{file}"
+      filename = ''
+      begin
+        connect
+        filename = "#{SecureRandom.urlsafe_base64(6)}_#{Time.now.to_i}.csv"
+        blob_name = @path ? "#{file}" : "#{@path.delete_suffix('/')}/#{file}"
```

Comment:
is it reversed?

Reply:
Yes, the logic code is reverse
I fixed its.